app/src/utils/searchManager.tsx

```diff
@@ -64,13 +64,24 @@ abstract class BaseSearchManager implements SearchManager {
   }
 
   handleAddFilter(newFilter: Filter) {
-    const updatedFilters = [...this.currentFilters, newFilter];
+    const updatedFilters = this.currentFilters.map((filter) =>
+      filter.filter === newFilter.filter ? newFilter : filter
+    );
+
+    const filterExists = this.currentFilters.some(
+      (filter) => filter.filter === newFilter.filter
+    );
+
+    if (!filterExists) {
```

Comment:
Consider making `this.currentFilters` or maybe just updatedFilters a `Set` to not have to validate the value already exists before adding it. You can then convert it back to an array as `this.currentFilters = [...updatedFiltersSet];`